src/test/java/com/ning/http/client/async/BasicProxyAuthorizationTest.java

```diff
@@ -0,0 +1,168 @@
+        port2 = findFreePort();
+
+        ServerConnector listener = new ServerConnector(server);
+        listener.setHost(LOCALHOST);
+        listener.setPort(port1);
+        server.addConnector(listener);
+
+        HttpConfiguration https_config = new HttpConfiguration();
+        https_config.setSecureScheme("https");
+        https_config.setSecurePort(port2);
+        https_config.setOutputBufferSize(32768);
+
+        SslContextFactory sslContextFactory = new SslContextFactory();
+        ClassLoader cl = getClass().getClassLoader();
+        URL cacertsUrl = cl.getResource("ssltest-cacerts.jks");
+        String trustStoreFile = new File(cacertsUrl.toURI()).getAbsolutePath();
+        sslContextFactory.setTrustStorePath(trustStoreFile);
+        sslContextFactory.setTrustStorePassword("changeit");
+        sslContextFactory.setTrustStoreType("JKS");
+
+        log.info("SSL certs path: {}", trustStoreFile);
+
+        URL keystoreUrl = cl.getResource("ssltest-keystore.jks");
+        String keystoreFile = new File(keystoreUrl.toURI()).getAbsolutePath();
+        sslContextFactory.setKeyStorePath(keystoreFile);
+        sslContextFactory.setKeyStorePassword("changeit");
+        sslContextFactory.setKeyStoreType("JKS");
+
+        log.info("SSL keystore path: {}", trustStoreFile);
+
+        SecureRequestCustomizer src = new SecureRequestCustomizer();
+        src.setStsMaxAge(2000);
+        src.setStsIncludeSubDomains(true);
+        https_config.addCustomizer(src);
+
+        ServerConnector connector = new ServerConnector(server2,
+            new SslConnectionFactory(sslContextFactory, HttpVersion.HTTP_1_1.asString()),
+            new HttpConnectionFactory(https_config));
+        connector.setHost(LOCALHOST);
+        connector.setPort(port2);
+
+        server2.addConnector(connector);
+
+        server.setHandler(configureHandler());
+        server.start();
+
+        server2.setHandler(new EchoHandler());
+        server2.start();
+        log.info("Local HTTP server started successfully");
+    }
+
+    @Test
+    public void connectionProxyAuthorizationHeaderProperlySetTest() throws Throwable {
+        ProxyServer proxyServer = new ProxyServer(LOCALHOST, port1);
+
+        AsyncHttpClientConfig asyncHttpClientConfig = new AsyncHttpClientConfig.Builder()
+            .setAcceptAnyCertificate(true)
+            .setProperProxyAuthentication(true)
+            .build();
+
+        try (AsyncHttpClient client = getAsyncHttpClient(asyncHttpClientConfig)) {
+            Request request = client.prepareGet(getTargetUrl2()).setProxyServer(proxyServer).build();
+            Response response = client.executeRequest(request).get();
+
+            Assert.assertEquals(response.getStatusCode(), 200);
+            Assert.assertEquals(response.getHeader(CONNECTION_HEADER), KEEP_ALIVE);
+        }
+    }
+
+    @Test
+    public void connectionProxyAuthorizationHeaderNotProperlySetTest() throws Throwable {
```

Comment:
I would remove this test as we don't want to test an incorrect behavior (event with the system property). The system property is only a fail safe mechanism.